### PR TITLE
Add new command of 'add github labels to REPO' to bot (WIP)

### PR DESCRIPTION
**Pull request by halkeye**
WIP to support https://github.com/jenkinsci/.github/pull/27

### CODE REVIEW

**Comment by halkeye** [comment]
also need @bitwiseman's help to add support for updating labels, but waiting to hear back about the desired builder pattern for it.

**Comment by halkeye** [comment]
YAY! updating labels - https://github.com/github-api/github-api/pull/649

**Comment by halkeye** [comment]
IDEA: change this to rebase repo, and make sure we have github topics of "jenkins" and "jenkins-plugin" setup

**Comment by bitwiseman** [comment]
@halkeye  
See #724.   If you update to v1.109 you can give this a try.

File: pom.xml [hunk 1]
```diff
@@ -86,7 +86,7 @@
     <dependency>
       <groupId>org.kohsuke</groupId>
       <artifactId>github-api</artifactId>
-      <version>1.89</version>
+      <version>1.101</version>
```
Comment (bitwiseman) [inline pom.xml:89]:
```suggestion
      <version>1.109</version>
```

File: src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java [hunk 2]
```diff
@@ -801,7 +833,60 @@ private void renameGitHubRepo(Channel channel, User sender, String repo, String
...
+                } else if (ghLabels.containsKey(label.name)) {
+                    /*
+                     * when updateLabel is published
+                     * repo.updateLabel(
+                     *  Strings.isNullOrEmpty(label.oldname) ? label.name : label.oldname,
+                     *  label.name, // newname
+                     *  label.color,
+                     *  label.description
+                     * )
+                     */
```
Comment (bitwiseman) [inline src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java:874]: From https://github.com/github-api/github-api/pull/724 :
```suggestion
                     ghLabels.get(Strings.isNullOrEmpty(label.oldname) ? label.name : label.oldname)
                         .update()
                         .name(label.name) // newname
                         .color(label.color)
                         .description(label.description)
                         .done();
```

**Comment by slide** [comment]
Is this still a WIP?

**Comment by halkeye** [comment]
Yea I rebased it but it needs a permanent location for the yaml and tests

File: src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java [hunk 3]
```diff
@@ -77,6 +84,25 @@
 
     private final Random random = new Random(System.currentTimeMillis());
 
+    private static final RepoLabels labelsYAML;
+    static {
+        try {
+            URL url = new URL("https://raw.githubusercontent.com/jenkinsci/.github/4f59a6219705df14067688df4f2a503e143cfa92/labels.yaml");
```
Comment (timja) [inline src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java:90]: Point to master branch?

Comment (halkeye) [reply src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java:90]: master doesn't have the file so it would 404, but yea the plan would be master

**Comment by oleg-nenashev** [comment]
Would be nice to get it over the line to avoid further merge conflicts

**Comment by halkeye** [comment]
https://plugins-self-service-3ir4b.ondigitalocean.app/ is way easier

**Comment by oleg-nenashev** [comment]
Looks nice though the UI does not seem to work for me (blank screen, maybe
long initial data loading).
Time to start a thread in the infra mailing list?

**Pull request closed without merging**